vmaas_sync/vmaas_sync.py

```diff
@@ -281,19 +306,48 @@ def sync_cve_md():
             return True
 
 
+def _create_message_list(rows: list[tuple[str, str, str, str]]):
+    return [
+        {
+            "type": "re-evaluate_system",
+            "host": {"id": inventory_id, "org_id": org_id},
+            "timestamp": str(datetime.now(timezone.utc)),
+        }
+        for _, inventory_id, org_id, _ in rows
+    ]
+
+
+def _prepare_messages(updated_repos: dict[str, list[str]], rows: list[tuple[str, str, str, str]], repo_based: bool):
+    """
+    row: (repo_name, inventory_id, org_id, vmaas_json)
+    """
+    if not repo_based:
+        return _create_message_list(rows)
+
+    systems_to_reevaluate = []
+    for repo_name, inventory_id, org_id, vmaas_json in rows:
+        repo_packages = set(vmaas_json["package_list"])
+        #  intersection exists (there are some pkgs which changed) => reevaluate system
+        if set(updated_repos[repo_name]) & repo_packages:
```

Comment:
1. rename repo_packages -> system_packages ?
2. it won't work because `updated_repos` contains package names ("kernel"...), but `repo_packages` contains package NEVRAs ("kernel-0:1.0-1.x86_64"...)

  I'd probably used code from https://github.com/RedHatInsights/vmaas/blob/master/vmaas/common/rpm_utils.py#L24 for this